Source/WebInspectorUI/UserInterface/Base/Main.js

```diff
@@ -1825,11 +1844,27 @@ WI._updateWindowInactiveState = function(event)
 
 WI._windowResized = function(event)
 {
+    WI._updateLayoutMode();
     WI.tabBar.updateLayout(WI.View.LayoutReason.Resize);
-    WI._tabBrowserSizeDidChange();
     WI._updateSheetRect();
 };
 
+WI._updateLayoutMode = function()
+{
+    let computedMode = WI.LayoutMode.Default;
+    
+    if (WI.settings.enableNarrowLayoutMode.value && window.innerWidth < WI.NarrowLayoutMaximumWidth)
+        computedMode = WI.LayoutMode.Narrow;
+    
+    if (WI.layoutMode !== computedMode) {
+        WI.layoutMode = computedMode;
+        document.body.classList.toggle("narrow", WI.layoutMode === WI.LayoutMode.Narrow);
+        WI.detailsSidebar.updateLayout(WI.View.LayoutReason.Resize);
+    }
+    
+    WI._tabBrowserSizeDidChange();
```

Comment:
this definitely isn't the most ideal as it's kinda odd for us to always do something even though we have a "dont do anything unless things have changed" `if` right above, but i understand the reasoning behind this so i think it's fine (for now)

Reply:
Yeah, I agree this isn't ideal.